### PR TITLE
Add optional custom validation and auto height adjustment

### PR DESCRIPTION
**Pull request by DavidBanksNZ**
Opening a PR in our forked repository first to get internal feedback. I've tried to minimise my changes here, believe it or not ;)

There are some minor changes I think should remain, like making sure there aren't global variables all over the place like the current implementation has.

I've also added an npm script to perform js minification, and and accordingly a .gitignore file to make sure junk files & node modules don't get committed.

### CODE REVIEW

**Comment by DavidBanksNZ** [comment]
I will also need to update README & CHANGELOG before opening a PR back to the original author. But want to make sure everything else is fine before doing that.

File: materialize-stepper.js [hunk 1]
```diff
@@ -135,41 +138,71 @@ $.fn.nextStep = function(callback, activefb, e) {
...
       if(settings.autoFocusInput) step.find('input:enabled:visible:first').focus();
       $this.trigger('stepchange').trigger('step'+(step_num+1));
       if(step.data('event')) $this.trigger(step.data('event'));
       if(callback)callback();
+     
+      // Mutation observer to update height, if required
+      if (settings.autoAdjustHeight) {
+         var updateTimer;
+         var updateInterval = 100;
+         var observer = new MutationObserver(function() {
```
Comment (Paul-n4l) [inline materialize-stepper.js]: MutationObserver is not supported on IE10, and materializecss claims to support ie10. Will this at least degrade gracefully on ie10?

File: materialize-stepper.js [hunk 2]
```diff
@@ -241,18 +304,44 @@ $.fn.activateStepper = function(options) {
...
+
+function updateHeight ($stepper) {
+   var $active = $stepper.find('.step.active'),
+      scrollHeight = $active.find('.step-content').get(0).scrollHeight,
+      newHeight = scrollHeight + 84;
+   $stepper.css({'min-height': newHeight + 'px'});
+}
+
+function isHorizontal ($stepper) {
+   return window.innerWidth >= 993 && $stepper.hasClass('horizontal');
```
Comment (overstimulated) [inline materialize-stepper.js]: just wondering why it is specific to >=993. do you check for window orientation here?

**Comment by overstimulated** [comment]
Hi Guys! this one is still open :) haha

**Pull request merged**